toolchain/install/busybox_info_test.cpp

```diff
@@ -90,18 +91,18 @@ TEST_F(BusyboxInfoTest, SymlinkInCurrentDirectoryWithDot) {
   auto info = GetBusyboxInfo(target.string());
   ASSERT_TRUE(info.ok()) << info.error();
   EXPECT_THAT(info->bin_path, Eq(dir_ / "./carbon-busybox"));
-  EXPECT_THAT(info->mode, Eq("carbon"));
+  EXPECT_THAT(info->mode, Eq(std::nullopt));
 }
 
 TEST_F(BusyboxInfoTest, ExtraSymlink) {
   MakeFile(dir_ / "carbon-busybox");
-  MakeSymlink(dir_ / "carbon", "carbon-busybox");
-  auto target = MakeSymlink(dir_ / "c", "carbon");
+  MakeSymlink(dir_ / "c", "carbon-busybox");
+  auto target = MakeSymlink(dir_ / "carbon", "c");
 
   auto info = GetBusyboxInfo(target.string());
   ASSERT_TRUE(info.ok()) << info.error();
   EXPECT_THAT(info->bin_path, Eq(dir_ / "carbon-busybox"));
-  EXPECT_THAT(info->mode, Eq("carbon"));
+  EXPECT_THAT(info->mode, Eq(std::nullopt));
 }
```

Comment:
Would it make sense to add a new test which tests the original symlink order, and that "c" is now the result? i.e., something like `OriginalSymlinkNameRetained`

Reply:
Yeah, this is implied below but much better to have a focused test for it here.